### PR TITLE
Fix deleting paths before the import to support directories

### PR DESCRIPTION
**Pull request by wojtekn**
## Related issues

- Fixes https://github.com/Automattic/dotcom-forge/issues/9220

## Proposed Changes

I propose fixing deleting paths before the import to ensure default plugins and themes are deleted before we import the site content.

## Testing Instructions

1. Import sites from different formats (Jetpack Backup, Playground ZIP file, .wpress file, Local export)
2. Confirm that default set of plugins themes that are included in clean WordPress are not visible on imported site if they were not included in the site being imported

## Pre-merge Checklist

- [ ] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

File: src/lib/import-export/import/importers/importer.ts
Comment (wojtekn) [inline src/lib/import-export/import/importers/importer.ts:115]: I renamed the method and parameter to show better what it does.

File: src/lib/import-export/import/importers/importer.ts
Comment (wojtekn) [inline src/lib/import-export/import/importers/importer.ts:117]: It supports deleting file and directory recursively.

**Comment by nightnei** [comment]
`Import sites from different formats`
Could you please add more details of what means `different formats`? It would be handy to understand how I can create different types that are necessary for tests.

>Confirm that core plugins like Hello Dolly or Akismet and themes like 2020 are not installed if they are not included in the site being imported

Should I install the 2020 theme and then assert that it's reverted to some default theme and two default plugins were removed? Also I tried to install it and can't find 2020 theme, hm.

Sorry, I didn't work with exporting/importing before, and actually I worked with plugins just a little bit. So such short testing steps are very difficult for me to understand.

**Comment by wojtekn** [comment]
> Import sites from different formats
> Could you please add more details of what means different formats? It would be handy to understand how I can create different types that are necessary for tests.

It seems we haven't [documented all formats in docs yet](https://developer.wordpress.com/docs/developer-tools/studio/#import-export). We support:
- Jetpack Backup file
- Playground ZIP file
- .wpress exported from [All-in-One WP Migration and Backup](https://wordpress.org/plugins/all-in-one-wp-migration/)
- Local by Flywheel export file

> Should I install the 2020 theme and then assert that it's reverted to some default theme and two default plugins were removed? Also I tried to install it and can't find 2020 theme, hm.

Create a vanilla WordPress site in Studio and see what plugins and themes are included by default. Then, when you add a new site by importing it from the file, confirm that those default plugins and themes are not included in that site, if they were not a part of export file.

Also see related issue for more context on use case reported by user.

**Review by katinthehatsite — APPROVED** [review]
It works as expected for me and the default plugins get deleted 👍 

Please note that I tested only with the `wpress` file and the Jetpack backup so perhaps we can wait for another review.

**Comment by nightnei** [comment]
@wojtekn 
### Jetpack Backup:

<table>
<tr>
<td>wordpress.com (8 plugins)
<td>trunk (9 plugins - additionally I see "Hello Dolly")
<td>this branch (8 plugins)
</tr>
<tr>
<td>Screenshot 2024-11-15 at 12 36 36
<td>Screenshot 2024-11-15 at 12 36 59
<td>Screenshot 2024-11-15 at 12 37 25

</tr>
</table>

So the difference is only "Hello Dolly", is it what we expect, that "Hello Dolly" is ignored during importing of "Jetpack Backup"? So actually we are ignoring not all default plagins, but only those which are extra for wordpress.com?

I will proceed to test other 3 types of imports.

**Comment by wojtekn** [comment]
> So actually we are ignoring not all default plagins, but only those which are extra for wordpress.com?

@nightnei in most straightforward words:
- before the fix, import added plugins from the backup file to the Site in Studio
- after the fix, import replaces plugins in the Studio site with those from the backup file

**Comment by nightnei** [comment]
>@nightnei in most straightforward words:
>before the fix, import added plugins from the backup file to the Site in Studio
>after the fix, import replaces plugins in the Studio site with those from the backup file


@wojtekn  got it, thanks you. It's really much more straightforward and clear to understand what we are actually doing.
Tested with:
1) created website, added a few plugins
2) Importad Jetpack Backup site
3) Confirmed that all plugins were replaced by plugins from imported "Jetpack Backup":
<table>
<tr>
<td>BEFORE IMPORT (12 items)
<td> AFTER IMPORT (8 items)
</tr>
<tr>
<td>Screenshot 2024-11-15 at 13 02 27
<td>Screenshot 2024-11-15 at 13 07 11
<tr>
</table>

### All-in-One WP Migration and Backup
Tested [All-in-One WP Migration and Backup](https://wordpress.org/plugins/all-in-one-wp-migration/) - all good, the same result as with "Jetpack Backup".

### Local
[Local](https://localwp.com/) export - all good, the same result what I have in local - no plugins installed at all:<br />
Screenshot 2024-11-15 at 14 02 32

### Playground
[Playground](https://playground.wordpress.net/) .zip file - the same result as for Local - no plugins installed


It's really good PR/fix - if we are importing a site, we should replace all plugins too 👍

**Review by nightnei — APPROVED** [review]
Looks good and test well (look previous comment ^) 👍

**Comment by wojtekn** [comment]
Thanks for detailed testing @nightnei  and @katinthehatsite !

**Pull request merged**
